### PR TITLE
Feature/other xserver cleanup

### PR DESCRIPTION
**Pull request by sunweaver**
This PR cleans up references to other XServers unrelated to NX. Also the Xprint feature is completely dropped from the code as it is unused in NX.

### CODE REVIEW

**Comment by Ionic** [comment]
I'll need time to take a look at all of that, sorry.

**Comment by sunweaver** [comment]
Hi Ionic,

I'd highly appreciate a review as I have the next PR candidate pending (https://github.com/sunweaver/nx-libs/commit/f71ca7372af5558e8c7c913cc0f4bce82463cbd5).

File: nx-X11/programs/Imakefile [hunk 1]
```diff
@@ -1,8 +1,3 @@
-XCOMM $Xorg: Imakefile,v 1.4 2000/08/17 19:47:01 cpqbld Exp $
-
-
-
-XCOMM $XFree86: xc/programs/Imakefile,v 3.54 2003/04/14 20:37:16 herrb Exp $
```
Comment (Ionic) [inline nx-X11/programs/Imakefile:5]: Uhm, I'd actually leave that in there as a reference for when it was last changed? The dates in there have been "helpful" to see that nomachine replaced some files with newer or older versions of the same file.

File: nx-X11/programs/Imakefile [hunk 2]
```diff
@@ -23,156 +18,15 @@ XCOMM $XFree86: xc/programs/Imakefile,v 3.54 2003/04/14 20:37:16 herrb Exp $
...
-#if BuildServer || UseRgbTxt
-       RGBSRCDIR = rgb
-#endif
 #if BuildServer
         XSSRCDIR = Xserver
 #endif
-#if BuildXkbcomp
-      XKBCOMPDIR = xkbcomp
-#endif
-#ifndef Win32Architecture
```
Comment (Ionic) [inline nx-X11/programs/Imakefile:35]: Be careful here. Note that from this line on, #if[n]defs are doubly-nested. Where does this #ifndef Win32Architecture stop?

File: nx-X11/programs/Imakefile [hunk 3]
```diff
@@ -23,156 +18,15 @@ XCOMM $XFree86: xc/programs/Imakefile,v 3.54 2003/04/14 20:37:16 herrb Exp $
...
-     XHOSTSRCDIR = xhost
-#if BuildPlugin && BuildXaw
-       XRXSRCDIR = xrx
-#endif
-       XSMSRCDIR = xsm
-#if BuildXterm
-     XTERMSRCDIR = xterm
-#endif
-      SCRIPTSDIR = scripts
-#endif /* Win32Architecture */
```
Comment (Ionic) [inline nx-X11/programs/Imakefile:86]: #ifndef Win32Architecture ends here. Ok, everything removed.

File: nx-X11/programs/Xserver/Imakefile [hunk 4]
```diff
@@ -562,13 +352,9 @@ NXWIN = NXWin
 #endif
 #if (defined(XF86Server) && XF86Server) || \
     (defined(XorgServer) && XorgServer) || \
-    (defined(XnestServer) && XnestServer) || \
     (defined(NXAgentServer) && NXAgentServer) || \
-    (defined(XVirtualFramebufferServer) && XVirtualFramebufferServer) || \
-    (defined(XdmxServer) && XdmxServer) || \
-    (!MakeDllModules && defined(XWinServer) && XWinServer) || \
     (!MakeDllModules && defined(NXWinServer) && NXWinServer)
```
Comment (Ionic) [inline nx-X11/programs/Xserver/Imakefile:356]: What is "NXWinServer" really? Does it depend on XWinServer?

File: nx-X11/programs/Xserver/Imakefile [hunk 5]
```diff
@@ -1423,127 +954,12 @@ ServerTarget(NXWin,$(NXWINDIRS),$(NXWINOBJS), \
...
-#endif /* DarwinQuartzSupport */
-
-#endif /* XDarwinServer */
-
-
 CFBSUBDIRS = $(CFB8SUBDIR) $(CFB16SUBDIR) $(CFB24SUBDIR) $(CFB32SUBDIR)
 MIEXTDIRS = $(SHADOWDIR) $(LAYERDIR) $(ROOTLESSDIR) $(MIDAMAGEDIR)
 IPLANDIRS = $(IPLAN2P2DIR) $(IPLAN2P4DIR) $(IPLAN2P8DIR)
 DDXDIRS = $(DECWSDDXDIR) $(SUNDDXDIR) $(LYNXDDXDIR) \
-	  $(HPDDXDIR) $(XFREE86DDXDIR) $(XWINDDXDIR) $(DARWINDDXDIR) \
```
Comment (Ionic) [inline nx-X11/programs/Xserver/Imakefile:1545]: DARWINDDXDIR is undefined now.

**Comment by sunweaver** [comment]
Ionic, I fixed the DARWINDDXDIR is undefined issue and force-pushed.

**Pull request merged**